src/basic/process-util.c

```diff
@@ -1727,6 +1728,36 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, pid_t *ret_pid) {
+        _cleanup_(posix_spawnattr_destroy) posix_spawnattr_t attr;
+        sigset_t mask;
+        pid_t pid;
+        int r;
+
+        assert(path);
+        assert(argv);
+        assert(ret_pid);
+
+        assert_se(sigemptyset(&mask) >= 0);
+        assert_se(sigaddset(&mask, SIGCHLD) >= 0);
+
+        assert_se(posix_spawnattr_init(&attr) == 0);
+        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGMASK);
```

Comment:
Isn't that equivalent to what happened before with fork/exec? Or did I misunderstand it?

Reply:
or let's put it another way - by default if nothing is passed it will inherit pid 1's sigmask. Is that what we want? Or do we want to pass something?

Reply:
I want to keep behaviour the same, if masking sigchld is the wrong thing then I'll change it. What should it do here? If I pass nothing, it will inherit from pid1. If that's the right thing then I'll pass nothing. If it's not, what should be the mask of the new process?